### ISSUE TITLE
[FEATURE] enhanced shell completions

### ISSUE THREAD

**Issue by brandonbloom** [deref/exo#285]
As of #213, we have support for _static_ shell completions. There are a few enhancements to make:

1) more robust install support for linux and windows
2) _automatic_ installation as part of our install script
3) _dynamic_ completions, for example, completing component names on commands like `exo start ...`

This is all relatively low priority, but (3) in particular is probably a good "help wanted" feature for new commers!

**Comment by brandonbloom** [comment]
Let's say that this is blocked by #301.

**Comment by MaxmumP** [comment]
I had difficulty getting completions to generate on Fish. I have no clue why, as `pkg-config --variable completionsdir fish` seems to resolve the proper directory, and `exo completion generate fish` works fine to output a completions file (I was able to just pipe that where it belonged). I'm not sure if this should be a new issue or could be attached to this one, but I'll start here.

I wrote up my solution (to help others) here: https://stackoverflow.com/questions/70010126/generating-completions-file-for-exo-in-fish/70010127#70010127
I tried to tag it as exo but I don't have 1500 reputation so couldn't create the tag.

**Comment by brandonbloom** [comment]
cc @BenElgar who can help with installer and shells stuff

**Comment by BenElgar** [comment]
I'm hoping #528 improves the fish use case.

**Comment by brandonbloom** [comment]
Thanks @BenElgar!

@MaxmumP, we'll let you know when this lands in the next release & maybe you can give it a try and let us know how it works. Thanks for the report.

**Comment by MaxmumP** [comment]
Yeah, that looks great. I had assumed you had already been using pkg-config to find the completions directory. Was impressed at how fast @BenElgar got that in! :-)

**Comment by brandonbloom** [comment]
This was prematurely closed along with the Fish fix. Dynamic completions are not yet implemented.
